app/Http/Controllers/LinkController.php

```diff
@@ -51,16 +51,23 @@ public function performRedirect(Request $request, $short_url, $secret_key=false)
         $link = Link::where('short_url', $short_url)
             ->first();
 
-        // Return 404 if link not found
+        // Return an error if the link does not exist
+        // or return a 404 if SETTING_REDIRECT_404 is set to true
         if ($link == null) {
-        	return abort(404);
+            if (env('SETTING_REDIRECT_404')) {
+                return response(view("errors.404"), 404);
+            }
+            
+            return view('error', [
```

Comment:
I can add the 404 status code to these error messages. Currently they do not redirect, are you suggesting to also remove the SETTING_REDIRECT_404 env check?